### PR TITLE
ROASTER-135: Fix wildcard handling in Types.toResolvedType

### PR DESCRIPTION
**Pull request by candiru**
This commit fixes the handling of types like List<? extends Number> in Types.toResolvedType.

### CODE REVIEW

**Comment by gastaldi** [comment]
Interesting. Can you provide a test case as well?

**Comment by candiru** [comment]
I'm not sure if it's okay to add EasyMock as a test dependency, let me just copy the test from the issue (https://issues.redhat.com/browse/ROASTER-135):

```java
@Test
public void toResolvedType()
{
   Importer<?> importer = EasyMock.createMock(Importer.class);

   EasyMock.expect(importer.getImport("List<? extends Account>")).andReturn(null).once();
   EasyMock.expect(importer.getImport("?extendsAccount")).andReturn(null).once();
   EasyMock.replay(importer);

   assertEquals("List<?extendsAccount>", toResolvedType("List<? extends Account>", importer));
   // should be: assertEquals("List<? extends Account>", toResolvedType("List<? extends Account>", importer));
}
```

So this demonstrates the issue, the "should be" line passes after the fix.

What do you think?

**Comment by gastaldi** [comment]
I think it's fine but I'd prefer to use Mockito instead

**Comment by candiru** [comment]
Sure, I added the test with Mockito.

File: api/src/main/java/org/jboss/forge/roaster/model/util/Types.java [hunk 1]
```diff
@@ -312,7 +312,13 @@ public static String toResolvedType(String type, Importer<?> importer)
          {
             genericJoiner.add(toResolvedType(genericPart, importer));
          }
-         builder.append("<").append(genericJoiner.toString()).append(">");
+
+         String genericParts = genericJoiner.toString();
+         genericParts = genericParts.replace("?extends", "? extends ");
```
Comment (gastaldi) [inline api/src/main/java/org/jboss/forge/roaster/model/util/Types.java:317]: Maybe it's better to perform the replace inside the for loop or make sure that `splitGenerics` returns the expected output

**Comment by gastaldi** [comment]
Can you squash your commits also?

**Comment by candiru** [comment]
The replace now happens in the loop as you suggested, and I squashed all three commits (with force push).

I noticed that my branch has a wrong name (ROASTER-123 instead of ROASTER-135), if you think that's a problem, let me know and I'll create another pull request from a new branch.

**Comment by gastaldi** [comment]
No, that's not a problem. Looks good, I'll merge. Thank you!

**Comment by candiru** [comment]
Great, thank you!

**Pull request merged**